### PR TITLE
Feature/project template

### PR DESCRIPTION
**Pull request by mjendza**
I created Net Core Template from DDD sample:
1. Changed namespaces
2. Sample app with controller + DDD with Issue entity
3. Hard Coded version of nugets - they should be changed manually
4. Template should be published as a nuget package
Howto install:
#install form directory
dotnet new -i "src\Template\Template.WebAPI"
#create project from template
dotnet new GoldenEye.WebAPI

### CODE REVIEW

File: src/Template/Template.WebAPI/Backend/Issues/Handlers/IssueCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+namespace Backend.Issues.Handlers
+{
+    internal class IssueCommandHandler
+        : ICommandHandler<CreateIssue>
+    {
+        private IEventBus eventBus;
+
+        public IssueCommandHandler( IEventBus eventBus )
+        {
+            this.eventBus = eventBus;
```
Comment (oskardudycz) [inline src/Template/Template.WebAPI/Backend/Issues/Handlers/IssueCommandHandler.cs]: Could you add check: `this.eventBus = eventBus  ?? throw new ArgumentException(nameof(eventBus));` It's good practice to have that, especially that is being used in other places.

File: src/Template/Template.WebAPI/Backend/Issues/Handlers/IssueCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        }
+
+        public async Task Handle(CreateIssue message, CancellationToken cancellationToken)
```
Comment (oskardudycz) [inline src/Template/Template.WebAPI/Backend/Issues/Handlers/IssueCommandHandler.cs]: Could you change parameter name from `message` to `command`? It'd be imho more descriptive.

File: src/Template/Template.WebAPI/Backend/Issues/Issue.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        }
+
+        public Issue(Guid id, IssueType type, string title)
+        {
+            Id = id;
+            Type = type;
+            Title = title;
+        }
+
+        public Guid StreamId => Id;
```
Comment (oskardudycz) [inline src/Template/Template.WebAPI/Backend/Issues/Issue.cs]: Could you move this declaration under properties and above constructors? :)

File: src/Template/Template.WebAPI/Backend/Issues/Issue.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using Backend.Contracts.Issues;
+using GoldenEye.Backend.Core.DDD.Events;
+using GoldenEye.Shared.Core.Objects.General;
+
+namespace Backend.Issues
+{
+    public class Issue : IEvent, IHasId
```
Comment (oskardudycz) [inline src/Template/Template.WebAPI/Backend/Issues/Issue.cs]: This class is not an event, it's aggregate, so it shouldn't implement the `IEvent` interface, it should implement `IAggregate` thou

File: src/Template/Template.WebAPI/Backend/Issues/Projections/IssueProjection.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+{
+    internal class IssueProjection : ViewProjection<Contracts.Issues.Views.IssueView, Guid>
+    {
+        public IssueProjection()
+        {
+            ProjectEvent<IssueCreated>(ev => ev.IssueId, (item, @event) => item.Apply(@event));
+        }
+
+        private void Apply(Contracts.Issues.Views.IssueView item, IssueCreated @event)
+        {
```
Comment (oskardudycz) [inline src/Template/Template.WebAPI/Backend/Issues/Projections/IssueProjection.cs]: Probably we should have some mapping here

File: src/Template/Template.WebAPI/Template.WebAPI.csproj [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    <RootNamespace>GoldenEye.WebAPI</RootNamespace>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Content Include=".template.config\template.json" />
+  </ItemGroup>
+  
+ <ItemGroup>
+	<PackageReference Include="AutoMapper" Version="6.2.2" />
+    <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="3.2.0" />
```
Comment (oskardudycz) [inline src/Template/Template.WebAPI/Template.WebAPI.csproj]: Could you update packages to latest versions?

File: src/Template/GoldenEye.WebApi.SimpleDDD/Backend/Issues/Issue.cs [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+        public Guid Id { get; private set; }
+
+        public IssueType Type { get; private set; }
+
+        public string Title { get; private set; }
+
+        public string Description { get; private set; }
+
+        public Guid StreamId => Id;
```
Comment (oskardudycz) [inline src/Template/GoldenEye.WebApi.SimpleDDD/Backend/Issues/Issue.cs]: This should be removed, it's leftover from the `IEvent` interface implementation

File: src/Template/GoldenEye.WebApi.SimpleDDD/Backend/Issues/Projections/IssueProjection.cs [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+{
+    internal class IssueProjection : ViewProjection<IssueView, Guid>
+    {
+        public IssueProjection()
+        {
+            ProjectEvent<IssueCreated>(ev => ev.IssueId, (item, @event) => item.Apply(@event));
+        }
+
+        private void Apply(IssueView item, IssueCreated @event)
+        {
```
Comment (oskardudycz) [inline src/Template/GoldenEye.WebApi.SimpleDDD/Backend/Issues/Projections/IssueProjection.cs]: Please provide here mapping of values from `@event` object to `item`

File: src/Template/GoldenEye.WebApi.SimpleDDD/Startup.cs [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+using GoldenEye.Backend.Core.WebApi.Modules;
+using GoldenEye.Backend.Core.WebApi.Registration;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Backend.DDD.WebApi.Sample
+{
+    public class Startup
```
Comment (oskardudycz) [inline src/Template/GoldenEye.WebApi.SimpleDDD/Startup.cs]: See new improved `Startup.cs` in https://github.com/oskardudycz/GoldenEye/pull/58/files#diff-49cb8c2ec8c71b61bd15b221102c7dd6

File: src/Template/GoldenEye.WebApi.SimpleDDD/Backend/Module.cs [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+using GoldenEye.Backend.Core.Marten.Registration;
+using GoldenEye.Shared.Core.Modules;
+using Marten;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using IssueContracts = Contracts.Issues;
+
+namespace Backend
+{
+    public class Module : ModuleBase
```
Comment (oskardudycz) [inline src/Template/GoldenEye.WebApi.SimpleDDD/Backend/Module.cs]: Please merge laterst master, currently `ModuleBase` was renamed to `Module`. See more in https://github.com/oskardudycz/GoldenEye/pull/58/files#diff-ebb3e38110765db415500e25d6cc4482

**Comment by oskardudycz** [comment]
@mjendza sure, I'll try to have a look on that this week

File: src/Template/GoldenEye.WebApi.SimpleDDD/Backend/Issues/Handlers/IssueQueryHandler.cs [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+        public IssueQueryHandler(IReadonlyRepository<IssueView> repository)
+        {
+            this.repository = repository ?? throw new ArgumentException(nameof(repository));
+        }
+
+        public async Task<IReadOnlyList<IssueView>> Handle(GetIssues message, CancellationToken cancellationToken)
+        {
+            return await repository
+                .GetAll()
+                //.ProjectTo<IssueView>()
```
Comment (oskardudycz) [inline src/Template/GoldenEye.WebApi.SimpleDDD/Backend/Issues/Handlers/IssueQueryHandler.cs]: this could be removed

**Comment by mjendza** [comment]
To create nuget we need to use nuspec can you setup it for us :) Please @oskardudycz 
``<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
  <metadata>
    <id>GarciaSoftware.ConsoleTemplate.CSharp</id>
    <version>1.0.0</version>
    <description>
      Creates the Garcia Software console app.
    </description>
    <authors>Catalina Garcia</authors>
    <packageTypes>
      <packageType name="Template" />
    </packageTypes>
  </metadata>
</package>``

**Pull request merged**